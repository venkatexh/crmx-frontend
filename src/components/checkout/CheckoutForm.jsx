import {PaymentElement, useElements, useStripe} from "@stripe/react-stripe-js";
import {useSelector} from "react-redux";

const CheckoutForm = () => {
  const state = useSelector(({loggedUser}) => ({loggedUser}));
  const {firstName, lastName, email, billing_address} = state.loggedUser;
  const stripe = useStripe();
  const elements = useElements();
  const options = {
    wallets: {
      googlePay: 'auto'
    },
  }
  const handleSubmit = async (event) => {
    event.preventDefault();
    if (!stripe || !elements) {
      return;
    }
    const result = await stripe.confirmPayment({
      elements,
      confirmParams: {
        return_url: "http://localhost:3000/payment-success-redirect",
        payment_method_data: {
          billing_details: {
            name: `${firstName} ${lastName}`,
            email,
            address: billing_address
          }
        }
      },
    });
    if (result.error) {
      console.log(result.error.message);
    } else {
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <PaymentElement options={options}/>
      <button disabled={!stripe} className={'payButton'}>Pay Now</button>
    </form>
  );
}

export default CheckoutForm;