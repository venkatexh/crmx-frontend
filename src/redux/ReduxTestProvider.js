import { Provider } from "react-redux";

const ReduxTestProvider = ({ children, reduxStore }) => (
  <Provider store={reduxStore}>{children}</Provider>
);

export default ReduxTestProvider;
