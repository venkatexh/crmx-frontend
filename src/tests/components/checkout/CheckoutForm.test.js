import { shallow } from "enzyme";
import React from "react";
import CheckoutForm from "../../../components/checkout/CheckoutForm";
import ReduxTestProvider from "../../../redux/ReduxTestProvider";

it("Checkout form component snapshot", () => {
  expect(
    shallow(
      <ReduxTestProvider>
        <CheckoutForm />
      </ReduxTestProvider>
    )
  ).toMatchSnapshot();
});
