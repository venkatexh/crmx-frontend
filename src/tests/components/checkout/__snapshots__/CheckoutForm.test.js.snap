// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout form component snapshot 1`] = `ShallowWrapper {}`;
